src/hotspot/share/gc/shenandoah/shenandoahGeneration.hpp

```diff
@@ -111,6 +111,16 @@ class ShenandoahGeneration : public CHeapObj<mtGC> {
   // Used by concurrent and degenerated GC to reset regions.
   void prepare_gc(bool do_old_gc_bootstrap);
 
+  // Compute evacuation budgets prior to choosing collection set.
```

Comment:
Do these methods need to be public? They looked more like helper methods in the way they are currently being used.

Another thing to think about is whether the reference parameters in one signature and regular ones in the other are potential sources of confusion or error in the future. I can see why they are like they are in their current use, but might indicate that perhaps a cleaner refactoring of the methods and the state that it uses or modifies, and where that state should reside.

Nothing to do at the moment, but worth thinking about.

Reply:
Agree with suggestion to make compute_evacuation_budgets and adjust_evacuation_budgets private.  Done.

I welcome further refactoring improvements in the future.  Am eager to get this code integrated now (as is) so that we can proceed with other performance improvements.